### DISCUSSION TITLE
What is joinlines for?

### DISCUSSION THREAD

**Discussion by garaiza-93** [mrcjkb/rustaceanvim#118] [Q&A] [answered]
I'm not understanding how this is different than `J`. Looking at `rustacean/commands/joinlines.lua`, it sends a request to rust-analyzer, but for what?

**Comment by mrcjkb** [comment, marked as answer]
Hey 👋 

Join lines is a bit smarter than J. For example, it can remove redundant braces, etc.
See [the rust-analyzer manual](https://rust-analyzer.github.io/manual.html#join-lines) and their demo gif:

![](https://user-images.githubusercontent.com/1711539/124515923-4504e800-dde9-11eb-8d58-d97945a1a785.gif)

**Reply by 9mm** [reply]
@mrcjkb hey i was wondering is there a reason visual line mode doesnt work with join lines? The only way i can get it to work is if i join every single line separately calling it over and over?

**Reply by mrcjkb** [reply]
Should be possible to implement.
I've created an issue for it https://github.com/mrcjkb/rustaceanvim/issues/335

**Reply by 9mm** [reply]
thanks!
